worker/sshserver/server.go

```diff
@@ -92,7 +99,7 @@ func NewServerWorker(config ServerWorkerConfig) (worker.Worker, error) {
 		s.config.Listener = listener
 	}
 
-	listener, closeAllowed := NewSSHServerListener(s.config.Listener)
+	listener := config.NewSSHServerListener(s.config.Listener, time.Second*10)
```

Comment:
This can look confusing, because time.Second*10 isn't used by our:
```go
// NewSSHServerListener returns a listener based on the given listener.
func NewSSHServerListener(l net.Listener, t time.Duration) net.Listener {
	return l
}
```
plugged into the manifold, by setting it to 10 seconds it kind of makes it look like it's used. Maybe 0 would be better and a doc explaining that the duration in this signature is purely for testing and as such we're just putting 0 for now.


On second thought, check my next comment in manifold.go about removing it from manifold & worker.

Reply:
The common pattern is to set everything at the manifold and not check for nil at the worker. Every time it starts up you have to check for nil, just set it and forget about it.